src/translator/text-based-field.ts

```diff
@@ -76,9 +76,9 @@ export namespace TextBasedField {
     const createValidationObject = (element: TElement) => {
         const validation: IYupValidationRule[] = [];
 
-        if (element.required) {
+        if (element.required !== undefined) {
             validation.push({
-                required: true,
+                required: element.required,
```

Comment:
we should assign the validation only if the field is required, is this causing an issue?